arbitrum-docs/devs-how-tos/how-to-estimate-gas.mdx

```diff
@@ -0,0 +1,135 @@
+
+```
+TXFEES = P * (L2G + ((L1P * L1S) / P))
+```
+
+## Where do we get all this information from?
+
+We will use two resources available in Arbitrum: the [ArbGasInfo](https://arbiscan.io/address/0x000000000000000000000000000000000000006c) precompile and the [NodeInterface](https://developer.arbitrum.io/arbos/gas#nodeinterfacesol).
+
+- P (L2 Gas Price) ⇒ Price to pay for each gas unit. It starts at 0.1 gwei and can increase depending on the demand for network resources.
+    - Call `ArbGasInfo.getPricesInWei()` and get the sixth element
+    - Call `NodeInterface.GasEstimateL1Component()` and get the second element
+    - Call `NodeInterface.GasEstimateComponents()` and get the third element (`baseFee`)
+- L2G (Gas used on L2) ⇒ Gas used to compute the transaction on L2. This does not include the “*posting on L1”* part of the calculations. The value of L2G will depend on the transaction itself, but having the data of the transaction, we can calculate it as follows:
+    - Call `NodeInterface.GasEstimateComponents()` with the transaction data and subtract the second element (`gasEstimateForL1`, which estimates the L1 part of the fees) from the first (`gasEstimate`, which includes both the L1 and the L2 parts).
+- L1P (L1 price per byte of data) ⇒ Cost of posting 1 byte of data on L1:
+    - Call `ArbGasInfo.getPricesInWei()` and get the second element
+    - We can also call the following methods, but we would have to multiply the result by 16 to get the cost of 1 byte of data: `ArbGasInfo.getL1BaseFeeEstimate()`, `ArbGasInfo.getL1GasPriceEstimate()`, `NodeInterface.GasEstimateL1Component()` and get the third element, `NodeInterface.GasEstimateComponents()` and get the fourth element.
+- L1S (Size of data to be posted on L1, in bytes) ⇒ This will depend on the data of the transaction. Keep in mind that Arbitrum adds a fixed amount to this number to make up for the static part of the transaction, which is also posted on L1 (140 bytes).
+    - For Arbitrum Nova (AnyTrust), the size of the data is also a fixed value, as only the Data Availability Certificate is posted on L1, [as explained here](https://developer.arbitrum.io/inside-anytrust#data-availability-certificates).
+
+## An example of how to apply this formula in your code
+
+Finally, we show an example of how to get the values we just described and how to estimate the gas usage of a transaction in Javascript. We will use our [SDK](https://github.com/OffchainLabs/arbitrum-sdk) to connect to `ArbGasInfo` and `NodeInterface`.
+
+We first instantiate a factory object for ArbGasInfo and NodeInterface, using two methods from the SDK. `l2Provider` is a regular JSON RPC provider for the L2 network we are using, and both `ARB_GAS_INFO` and `NODE_INTERFACE_ADDRESS` are the addresses of the precompiles in said network.
+
+```ts
+// Instantiation the ArbGasInfo and NodeInterface objects
+const arbGasInfo = ArbGasInfo__factory.connect(
+    ARB_GAS_INFO,
```

Comment:
let's either hardcode the ArbGasInfo address or import it form the sdk (likewise with the NodeInterface address)